optaplanner-core/src/main/java/org/optaplanner/core/impl/localsearch/decider/multithreaded/MultiThreadedLocalSearchDecider.java

```diff
@@ -0,0 +1,407 @@
+        resultQueue = null;
+    }
+
+    private ExecutorService createThreadPoolExecutor() {
+        ThreadPoolExecutor threadPoolExecutor
+                = (ThreadPoolExecutor) Executors.newFixedThreadPool(moveThreadCount, threadFactory);
+        if (threadPoolExecutor.getMaximumPoolSize() < moveThreadCount) {
+            throw new IllegalStateException(
+                    "The threadPoolExecutor's maximumPoolSize (" + threadPoolExecutor.getMaximumPoolSize()
+                    + ") is less than the moveThreadCount (" + moveThreadCount + "), this is unsupported.");
+        }
+        return threadPoolExecutor;
+    }
+
+    @Override
+    public void decideNextStep(LocalSearchStepScope<Solution_> stepScope) {
+        int stepIndex = stepScope.getStepIndex();
+        resultQueue.startNextStep(stepIndex);
+        int selectingMoveIndex = 0;
+        for (Move<Solution_> selectingMove : moveSelector) {
+            operationQueue.add(new MoveEvaluationOperation<>(stepIndex, selectingMoveIndex, selectingMove));
+            selectingMoveIndex++; // Increment required before if check
+            if (selectingMoveIndex < selectedMoveBufferSize) {
+                // Fill the buffer so move evaluation can run freely in parallel
+                // For reproducibility, the selectedMoveBufferSize always need to be selected,
+                // even if some of those moves won't be evaluated or foraged
+                continue;
+            }
+            OrderByMoveIndexBlockingQueue.MoveResult<Solution_> result;
+            try {
+                result = resultQueue.take();
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                break;
+            }
+            if (stepIndex != result.getStepIndex()) {
+                throw new IllegalStateException("Impossible situation: the solverThread's stepIndex (" + stepIndex
+                        + ") differs from the result's stepIndex (" + result.getStepIndex() + ").");
+            }
+            Move<Solution_> foragingMove = result.getMove().rebase(stepScope.getScoreDirector());
+            int foragingMoveIndex = result.getMoveIndex();
+            LocalSearchMoveScope moveScope = new LocalSearchMoveScope<>(stepScope, foragingMoveIndex, foragingMove);
+            if (!result.isMoveDoable()) {
+                logger.trace("{}        Move index ({}) not doable, ignoring move ({}).",
+                        logIndentation, foragingMoveIndex, foragingMove);
+            } else {
+                moveScope.setScore(result.getScore());
+                boolean accepted = acceptor.isAccepted(moveScope);
+                moveScope.setAccepted(accepted);
+                logger.trace("{}        Move index ({}), score ({}), accepted ({}), move ({}).",
+                        logIndentation,
+                        foragingMoveIndex, moveScope.getScore(), moveScope.getAccepted(),
+                        foragingMove);
```

Comment:
Would it be better to rewrap the lines of these logging parameters?

Reply:
What do you mean with rewrap? As in logger.isTrace() ? The getScore() and getAccepted() methods are simple getters, they would be inlined normally.